packages/apollo-links/src/query.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import axios from 'axios';
+
+import { Agreement } from "./types";
+
+export async function POST<T>(url: string, body: Record<string, string | number | undefined>) {
+  const headers = { 'Content-Type': 'application/json' };
+  const res = await axios.post<T>(url, body, { headers });
+
+  return res.data;
+}
+
+export async function GET<T>(url: string) {
+  const headers = { 'Content-Type': 'application/json' };
+  const res = await axios.get<T>(url, { headers });
+
+  return res.data;
+}
+
+interface AgreementsResponse {
+  agreements: Agreement[];
+}
+
+export async function fetchAgreements(authUrl: string, projectChainId: string): Promise<Agreement[]> {
+  const agreementsURL = new URL(`/agreements/${projectChainId}`, authUrl);
```

Comment:
How does auth service know about the latest deployment of a network's dictionary?

Reply:
auth-service need to handle that config